aip/general/0193/aip.md

```diff
@@ -0,0 +1,159 @@
+# Errors
+
+Error handling is an important part of designing simple and intuitive APIs.
+Consistent error handling allows developers to know how to expect to receive
+errors, and to reduce boilerplate by having common error-handling logic, rather
+than being expected to constantly add verbose error handling everywhere.
+
+## Guidance
+
+Services **must** clearly distinguish successful responses from error responses
+by using appropriate HTTP codes:
+- Informational responses issued on a provisional basis while request processing continues **must** use HTTP status codes between 100 and 199.
+- Successful responses **must** use HTTP status codes between 200 and 299.
+- HTTP status codes between 300 and 399 **must** be used to indicate further action like URL redirection needs to be taken in order to complete the request.
+- Errors indicating a problem with the user's request **must** use HTTP status
+  codes between 400 and 499.
+- Errors indicating a problem with the server's handling of an valid request
+  **must** use HTTP status codes between 500 and 599.
+
+### Structure
+
+Error responses **should** conform to the following interface:
+
+```typescript
```

Comment:
why typescript when the existing aip.dev is using protobuf / OpenAPI? https://aip-dev.github.io/aip.dev/136.